Switch from poetry to uv

ewwong2 opened the pull request:
- Switch from poetry to uv
    - Update Dockerfile
    - Update README
- Format and lint with ruff

ewwong2 committed:
Switch from poetry to uv

- Format and lint with ruff

djbarajas commented on Dockerfile:
```diff
@@ -1,28 +1,22 @@
-COPY ./pyproject.toml ./poetry.lock* /app/
-
-RUN bash -c "poetry install --no-root"
+WORKDIR /app
```
Ive seen best practices not use workdir but we can discuss later and change if needed

djbarajas reviewed (approved):
looks good. might be useful to have a github workflow to check if we are formatted properly to merge

ewwong2 committed:
Adjust formatting, add pre-commit hook, and adjust Dockerfile

- Switch to single quotes for Ruff
- Change Ruff line length to 120
- Update README to include pre-commit and ruff steps
- Switch README to use prestart script
- Add basic ruff Github action
- Update Dockerfile to use cd instead of workdir

ewwong2 committed:
Update ruff action
